src/DataLib/fendl32B_retrofit/groupr_tools.py

```diff
@@ -0,0 +1,36 @@
+# Import packages
+from string import Template
+
+# Define constants
+NENDF = 20 # unit for endf tape
+NPEND = 21 # unit for pendf tape
+NGOUT1 = 0 # unit for input gout tape (default=0)
+NGOUT2 = 31 # unit for output gout tape (default=0)
+IGN = 17 # neutron group structure option (corresponding to Vitamin J)
+IGG = 0 # gamma group structure option
+IWT = 11 # weight function option (corresponding to Vitamin E)
+LORD = 0 # Legendre order
+NTEMP = 1 # number of temperatures (default=1)
+NSIGZ = 1 # number of sigma zeroes (default=1)
+IPRINT = 1 # long print option (0/1=minimum/maximum) --(default=1)
+ISMOOTH = 1 # switch on/off smoother operation (1/0, default=1=on)
+TEMP = 293.16 # temperature in Kelvin
+SIGZ = 0 # sigma zero values (including infinity)
+MFD = 3 # file to be processed
+MATD = 0 # next mat number to be processed
+
+# Create input file general template
+njoy_input = Template(
```

Comment:
I don't know the answer, but wonder if you could do the substitution of the fixed items above immediately, and separately do substitution of the variable stuff later?

That is, can substitutions be done in two rounds?

Reply:
I believe it can, but I'll test it out now and if it does work, push a new commit with that modification.

Reply:
Ok, I just made a new update that substitutes in the constants immediately. I had to use `safe_substitute` here because not all of the placeholders are being called, which produces a `str` instead of substituting back into a `Template`, so I have to convert back to a `Template` again, but it seems to work fine.